File: cpp/1310_子数组异或查询_medium.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        int n = queries.size();
        int m = arr.size();
        vector<int> ans(n);
        vector<int> dict(m+1, 0);
        for (int i=1; i<m+1; i++)
            dict[i] = arr[i-1] ^ dict[i-1];
        for (int i=0; i<n; i++)
            ans[i] = dict[queries[i][0]] ^ dict[queries[i][1]+1];
        return ans;
    }
};

File: python/1310_子数组异或查询_medium.py
class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        n, m = len(arr), len(queries)
        mem, ans = [0] * (n+1), list()
        for i in range(1, n+1):
            mem[i] = arr[i-1] ^ mem[i-1]
        for i in range(m):
            ans.append(mem[queries[i][0]] ^ mem[queries[i][1] + 1])
        return ans
